app/controllers/DataSetController.scala

```diff
@@ -38,25 +41,28 @@ class DataSetController @Inject()(val messagesApi: MessagesApi) extends Controll
       (__ \ 'isPublic).read[Boolean]).tupled
 
 
-  def thumbnail(dataSetName: String, dataLayerName: String) = UserAwareAction.async { implicit request =>
+  def thumbnail(dataSetName: String, dataLayerName: String, w: Option[Int], h: Option[Int]) = UserAwareAction.async { implicit request =>
 
-    def imageFromCacheIfPossible(dataSet: DataSet) =
-    // We don't want all images to expire at the same time. Therefore, we add a day of randomness, hence the 1 day
-      Cache.get(s"thumbnail-$dataSetName*$dataLayerName") match {
+    def imageFromCacheIfPossible(dataSet: DataSet) = {
+      val width = Math.limit(w.getOrElse(DefaultThumbnailWidth), 1, MaxThumbnailHeight)
+      val height = Math.limit(h.getOrElse(DefaultThumbnailHeight), 1, MaxThumbnailHeight)
+      // We don't want all images to expire at the same time. Therefore, we add a day of randomness, hence the 1 day
```

Comment:
I know it's been like that before, but wouldn't the comment make more sense before line 58?